plugins/channel_track.go

```diff
@@ -196,7 +208,81 @@ func (p *ChannelTracker) nickCallback(b *seabird.Bot, m *irc.Message) {
 	//fmt.Printf("%s (%s) changed their name to %s\n", oldUser, p.uuids[newUser], newUser)
 }
 
-func (p *ChannelTracker) namesCallback(b *seabird.Bot, m *irc.Message) {
+func (p *ChannelTracker) modeCallback(b *seabird.Bot, m *irc.Message) {
+	// We only care about MODE messages where a specific user is
+	// changed.
+	if len(m.Params) < 3 {
+		return
+	}
+
+	logger := b.GetLogger()
+
+	channel := m.Params[0]
+	target := m.Params[2]
+
+	//change := m.Params[1]
+	//op := change[0]
+	//mode := change[1]
+
+	// Ensure we know about this user and this channel
+	u := p.LookupUser(target)
+	c := p.LookupChannel(channel)
+	if u == nil || c == nil {
+		logger.Warnf("Got MODE callback for %s on %s but we aren't tracking both", target, channel)
+		return
+	}
+
+	// Just send a WHO request and clear out the modes for this user because
+	// mode parsing is hard.
+	u.channels[channel] = make(map[rune]bool)
```

Comment:
are mode strings and who strings pretty different? if they're similar we could just have one parse line and parse it here as well. i'd prefer that instead of clearing out information we already have that might not be wrong.